linux_os/guide/system/ocp_idp_no_htpasswd/oval/shared.xml

```diff
@@ -0,0 +1,33 @@
+{{% set YAML_TEST_OVAL_VERSION = [5, 11] %}}
+
+{{% if target_oval_version >= YAML_TEST_OVAL_VERSION %}}
+<def-group>
+	<definition class="compliance" version="1" id="{{{ rule_id }}}">
+      <metadata>
+        <title></title>
+        {{{- oval_affected(products) }}}
+        <description>Sample rule</description>
+      </metadata>
+      <criteria operator="AND">
+        <criterion comment="Test that no element of spec.identityProviders has type htpasswd" test_ref="test_{{{ rule_id }}}"/>
+      </criteria>
+    </definition>
+
+    <ind:yamlfilecontent_test id="test_{{{ rule_id }}}" check_existence="none_exist" check="all" comment="No match of htpasswd in ..." version="1">
```

Comment:
Maybe the state is ignored because the test expects not objects to exist (`check_existence="none_exist`). The mere existence of the object is enough to result in fail, and the state doesn't need to evaluated.

Reply:
I see, thanks for the insight! So what about this - look for at least one object of the state that we don't want, and then negate the result of the test. Does this sound like a good way to have it done?

Reply:
I think so, some of our tests are written this way.

Reply:
I have rebased. The passing scenario now behaves as expected, but the failing scenario doesn't work - when there are two occurrences of `type` when one is OK and one is forbidden (and supposed to be matched by the state), the test concludes that the state doesn't match the object. Could you please help me with this? It's quite easy to test even locally as long as one passes the `--skip-valid` option.

Reply:
Set  set attribute `entity_check` of element `value_of` in the state to `at least one`:
`<ind:value_of check_existence="at least one" >htpasswd</ind:value_of>`

The default value of `entity_check` is `all`, [reference](https://github.com/OVALProject/Language/blob/7fa7bba7b48f09decb732d00b2be032a487ff9fc/docs/oval-definitions-schema.md#-entitystatesimplebasetype-).

So when there are multiple entities, it will only evaluate to `true` when all occurrences of `type` have value `htpasswd`.